### ISSUE TITLE
[Bug] Variables in the body with multipat form type.

### ISSUE THREAD

**Issue by Solovyov** [usebruno/bruno#352]
Variables are not inserted into the body with multipat form type.

mock return request body
![image](https://github.com/usebruno/bruno/assets/1317648/743d54ee-53a6-4ec6-a76d-124b1aef2f28)

**Comment by lared** [comment]
Seems similar to #329, I'll see if this is the same kind of quick and easy fix.

**Comment by lared** [comment]
Sorry, it's a bit more difficult fix - by the time we are interpolating things, we are already using Axios magic

```
{
  "method": "POST",
  "url": "http://google.com",
  "headers": {
    "content-type": "multipart/form-data; boundary=--------------------------197003115612960382152288"
  },
  "data": {
    "_overheadLength": 206,
    "_valueLength": 11,
    "_valuesToMeasure": [],
    "writable": false,
    "readable": true,
    "dataSize": 0,
    "maxDataSize": 2097152,
    "pauseStreams": true,
    "_released": false,
    "_streams": [
      "----------------------------197003115612960382152288\r\nContent-Disposition: form-data; name=\"whatever\"\r\n\r\n",
      "{{someVar}}",
      null,
      "----------------------------197003115612960382152288\r\nContent-Disposition: form-data; name=\"\"\r\n\r\n",
      "",
      null
    ],
    "_currentStream": null,
    "_insideLoop": false,
    "_pendingNext": false,
    "_boundary": "--------------------------197003115612960382152288"
  },
  "script": {},
  "vars": {},
  "assertions": [],
  "cancelToken": {
    "promise": {}
  }
}

```

To fix this we'd need to interpolate it first and *then* feed it through axios, but there is some conditional logic related to proxies and stuff in between. It's not *very* complicated, just requires more time than I have at the moment.

**Comment by adleatherwood** [comment]
I love Bruno's design and integration with projects/repos.  I would've incorporated it into my project today if I hadn't run into this.  Very much looking forward to this fix.

**Comment by Schoof-T** [comment]
Same here, looking forward to this fix!

**Comment by EmmanuelRoubion** [comment]
Yes, same here indeed: it would be much more elegant to be able to.

**Comment by t-jans** [comment]
Any progress on this issue?

**Comment by Its-treason** [comment]
> Any progress on this issue?

Not really. I try (or somebody else) to look into it the next few days. But i think it should be a relatively easy fix™.

**Comment by peauc** [comment]
Thanks for taking the time @Its-treason. I'll build it locally to try the fix 👀

**Comment by Schoof-T** [comment]
@peauc Any progress?

**Comment by dgiessing** [comment]
Just ran into this, looking forward to this being fixed

**Comment by Sebastian-Lerch** [comment]
Same here, I will probably find a workaround using axios directly however looking forward to the fix

**Comment by mstroppel** [comment]
Having same issue. You can do the following (.bru file):

```
...
headers {
  Content-Type: multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW
}

body:text {
  ------WebKitFormBoundary7MA4YWxkTrZu0gW
  Content-Disposition: form-data; name="grant_type"
  
  client_credentials
  ------WebKitFormBoundary7MA4YWxkTrZu0gW
  Content-Disposition: form-data; name="client_id"
  
  {{MyClientId}}
  ------WebKitFormBoundary7MA4YWxkTrZu0gW
  Content-Disposition: form-data; name="client_secret"
  
  {{MyAppSecret}}
  ------WebKitFormBoundary7MA4YWxkTrZu0gW
  Content-Disposition: form-data; name="resource"
  
{{MyResource}}
  ------WebKitFormBoundary7MA4YWxkTrZu0gW--
}
...
```

This example has 4 parts:

- grant_type
- client_id
- client_secret
- resource

**Comment by robert-smith-maersk** [comment]
For those looking for a workaround in the meantime, switching to body type 'Form URL Encoded' and adding the same params with variables there worked for me.

**Comment by helloanoop** [comment]
@lohxt1 Assigning this to you. Please get to speed on this issue and lets review @Its-treason PR if we can get to a solution that does not involve a breaking change

**Comment by xosealvarez** [comment]
Any updates?

**Comment by irharrier2** [comment]
> For those looking for a workaround in the meantime, switching to body type 'Form URL Encoded' and adding the same params with variables there worked for me.

This doesn't work for me.

**Comment by matthewb531** [comment]
Any updates please team? I'm using latest Bruno, no joy.

**Comment by zacharyjarnagin** [comment]
Also running into this issue. The work around to use "Form URL Encoded" won't work for me as I need to upload a file 😢 thanks to everyone working on this, [#2329](https://github.com/usebruno/bruno/pull/2329) seems promising!

**Comment by vagneurf** [comment]
I have the same issue, would love to have a fix or a workaround that allows to upload files while using variables

**Comment by adobitar** [comment]
Loving Bruno - this bug is painful for me right now.

**Comment by ejsmith** [comment]
Yes! This bug is really painful.

**Comment by ali-transmedia** [comment]
Is there any update on this? This bug is frustrating.

**Comment by nicklasbring8** [comment]
I'm also looking forward to a fix for this. It's very frustrating having to hardcode the variables in my multipart requests.

**Comment by helloanoop** [comment]
Hey everyone!

This is a priority for us to get it fixed in Sep 2024 (within the next 2 weeks).
@lohxt1 let's work on this after your work on #3094 is completed.

Since merging #2767, several issues that were hindering a solution to this problem now appear to be resolved.

I’m also open to introducing a breaking change, if necessary, to fully address this issue.

**Comment by ashishjain14** [comment]
Is this issue fixed?

**Comment by sirimhrzn** [comment]
It's been two weeks, are there any updates, we're planning to move to Bruno completely from Postman but this is really stopping us

**Comment by ali-transmedia** [comment]
When can this be fixed? @helloanoop. Some team members started returning to Postman due to this irritating bug.

**Comment by reggermont** [comment]
> It's been two weeks, are there any updates, we're planning to move to Bruno completely from Postman but this is really stopping us

Use v1.26.2, last version without the bug

**Comment by sirimhrzn** [comment]
Yeah but multipart form doesnot seem to work on it.
`Error invoking remote method 'send-http-request': Error: TypeError: Cannot read properties of undefined (reading 'value')`

**Comment by matthewb531** [comment]
Unless I'm mistaken... A fix for this has been merged ([yesterday](https://github.com/usebruno/bruno/pull/3142)) but I'm still seeing this buggy behaviour: <img width="695" alt="Screenshot 2024-10-01 at 08 46 01" src="https://github.com/user-attachments/assets/eb75b6cb-8bff-4d33-82e6-98d167293ae6">

**Comment by helloanoop** [comment]
This has been fixed and is available in `v1.32.0` 🎉

**Comment by olaf-lexemo** [comment]
Great, thanks a lot!

From: Anoop M D ***@***.***>
Date: Tuesday, 1 October 2024 at 19:54
To: usebruno/bruno ***@***.***>
Cc: Olaf ***@***.***>, Manual ***@***.***>
Subject: Re: [usebruno/bruno] [Bug] Variables in the body with multipat form type. (Issue #352)

This has been fixed and is available in v1.32.0 🎉
